packages/mongo-nav/src/mongo-select/MongoSelect.tsx

```diff
@@ -115,28 +115,61 @@ interface OrganizationMongoSelectProps extends BaseMongoSelectProps {
   isOnPrem?: boolean;
 }
 
-const onKeyDown: React.KeyboardEventHandler = e => {
-  // Stops default browser behavior from automatically scrolling the component
-  if ([keyMap.ArrowUp, keyMap.ArrowDown].includes(e.keyCode)) {
-    e.preventDefault();
-  }
-};
-
 function OrgSelect({
   current,
   data,
   urls,
   isActive,
-  onChange,
+  onChange: onChangeProp,
   onClick,
   constructOrganizationURL,
   isOnPrem,
   disabled,
   loading = false,
 }: OrganizationMongoSelectProps) {
+  const [renderedData, setRenderedData] = useState(data);
+  const [value, setValue] = useState('');
   const [open, setOpen] = useState(false);
   const onElementClick = useOnElementClick();
 
+  useEffect(() => {
+    setRenderedData(data);
+  }, [data]);
+
+  const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setValue(e.target.value);
+
+    if (onChangeProp) {
+      onChangeProp(e);
+      return;
+    }
+
+    const search = new RegExp(`${e.target.value}`, 'i');
```

Comment:
If you're using the same value (`e.target.value`) twice in a scope it's generally good practice to cache that value in a local variable.

This is a micro performance optimization, as every property access takes some (negligible) cost to do, but most importantly it guarantees the consistency of that value used throughout the function.

Such that you sanitize the value into a local variable at the beginning of the function and then delegate all usage to that one source of (consistent) truth.

Reply:
👍